ui/component/or-attribute-input/src/or-anomaly-config-chart.ts

```diff
@@ -0,0 +1,303 @@
+                || !(this.anomalyConfig as AnomalyDetectionConfigurationGlobal).minimumDatapoints ||!/^P(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$/.test((this.anomalyConfig as AnomalyDetectionConfigurationGlobal).timespan as string)){
+                this.errorMessage = i18next.t("anomalyDetection.invalidConfiguration");
+                return
+            }
+        }else if(this.anomalyConfig.type==="change"){
+            if((this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan === "" || !(this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan
+                || !(this.anomalyConfig as AnomalyDetectionConfigurationChange).minimumDatapoints || !/^P(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$/.test((this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan as string)){
+                this.errorMessage = i18next.t("anomalyDetection.invalidConfiguration");
+                return
+            }
+        }
+        if(this.attributeRef){
+            this._loading = true
+            const query = {
+                ids: [this.attributeRef.id],
+                select: {
+                    attributes: [
+                        this.attributeRef.name
+                    ]
+                },
+            } as AssetQuery;
+            try {
+                const response = await manager.rest.api.AssetResource.queryAssets(query);
+                this.assets = response.data || [];
+                if(this.attributeRef.name){
+                    if(this.assets[0].attributes){
+                        this.assetAttributes[0] = [0,this.assets[0].attributes[this.attributeRef.name]]
+                    }
+                }
+            } catch (e) {
+                console.error("Failed to get assets requested in settings", e);
+            }
+
+
+        }
+
+        if (this._data || !this.assetAttributes || !this.assets || (this.assets.length === 0 && !this.dataProvider) || (this.assetAttributes.length === 0 && !this.dataProvider)) {
+            this._loading = false
+            return;
+        }
+        this._loading = true;
+
+        this._startOfPeriod = 0;
+        this._endOfPeriod = 0;
+        if(this.anomalyConfig.type === "global" || this.anomalyConfig.type === "change"){
+            timespan =  moment.duration((this.anomalyConfig as AnomalyDetectionConfigurationGlobal | AnomalyDetectionConfigurationChange).timespan).asMilliseconds()
+            this._startOfPeriod = Date.now() - timespan * 5;
+            this._endOfPeriod = Date.now();
+        }else if(this.anomalyConfig.type === "forecast"){
+            timespan = 1000*60*20;
+            this._startOfPeriod = Date.now() - timespan * 2;
+            this._endOfPeriod = Date.now() + timespan;
+        }
+
+        this.datapointQuery = {
+            type: "all",
+            fromTimestamp: Date.now()- timespan * 5,
+            toTimestamp: Date.now()
```

Comment:
Maybe include some kind of "max time" on this, so it will only fetch for example 1 week max.